Assets/Scripts/BoardManager.cs

```diff
@@ -75,14 +218,129 @@ Vector3 RandomPosition()
         return RandomPosition;
     }
 
-    void LayoutObjectAtRandom(GameObject[] tileArray)
+    Vector3 RandomPositionNotCloseToInnerWall(List<Vector3> innerWallGridPositions)
     {
-        for (int i = 0; i < wallCount; i++)
+        int randomIndex = Random.Range(0, gridPositions.Count);
+        Vector3 RandomPosition = gridPositions[randomIndex];
+        if (!CheckIfPositionIsNextToPositions(RandomPosition, innerWallGridPositions))
         {
-            Vector3 randomPosition = RandomPosition();
-            GameObject tileChoice = tileArray[Random.Range(0, tileArray.Length)];
-            Instantiate(tileChoice, randomPosition, Quaternion.identity);
+            gridPositions.RemoveAt(randomIndex);
+            return RandomPosition;
+        }
+        return RandomPositionNotCloseToInnerWall(innerWallGridPositions);
+    }
+
+    bool CheckIfPositionIsNextToPositions(Vector3 position, List<Vector3> positions)
+    {
+        foreach (var pos in positions)
+        {
+            int diffX = (int)Math.Abs((float)(pos.x - position.x));
+            int diffY = (int)Math.Abs((float)(pos.y - position.y));
+
+            if (diffX + diffY <= 2 && diffX != 2 && diffY != 2)
```

Comment:
`diffX + diffY > 0 && diffX + diffY <= 2 && diffX != 2 && diffY != 2`
instead of 2 if